.github/workflows/ami-release-nix-single.yml

```diff
@@ -26,7 +26,11 @@ jobs:
       - name: Checkout Repo
         uses: actions/checkout@v3
         with:
-          ref: ${{ github.event.inputs.branch }}
+          repository: ${{ github.event.pull_request.head.repo.full_name || github.repository }}
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
+          fetch-depth: 0
+          fetch-tags: true
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
This looks duplicated in several places, which is too bad for maintenance. 

Github supports reusable workflows, can we deduplicate this using those? AFAICT it's only a matter of having another YAML file (say `shared-checkout`) which is then called in several places.

Reply:
@steve-chavez you are right. Let me see what I can do.